test_crates/function_param_types_and_return_value_types/src/lib.rs

```diff
@@ -0,0 +1,35 @@
+pub fn add(left: u64, right: u64) -> u64 {
+    left + right
+}
+
+pub fn fn_with_ref(param: &i32) -> i32 {
+    *param + 1
+}
+
+pub fn fn_with_mut_ref(param: &mut i32) {
+    *param += 1;
+}
+
+pub fn fn_with_generic<T: std::ops::Add<Output = T>>(a: T, b: T) -> T {
+    a + b
+}
+
+pub fn fn_with_generic_ref<T: std::ops::Add<Output = T>>(a: &T, b: &T) -> T
+where
+    T: Copy,
+{
+    *a + *b
+}
+
+pub fn fn_with_array_param(arr: [i32; 3]) -> i32 {
+    arr.iter().sum()
+}
+
+pub fn fn_with_qualified_param(p: std::option::Option<i32>) -> i32 {
+    p.unwrap_or(0)
+}
+
+use std::option::Option;
+pub fn fn_with_unqualified_param(p: Option<i32>) -> i32 {
+    p.unwrap_or(0)
+}
```

Comment:
nit: we always leave a trailing line at the end of files, to improve interoperability with terminal tools (there's an IDE setting for this in all IDEs if you want to not check it manually)
```suggestion
}

```